lib/Conversion/VerifToSMT/VerifToSMT.cpp

```diff
@@ -272,9 +274,19 @@ struct VerifBoundedModelCheckingOpConversion
       if (isa<seq::ClockType>(oldTy)) {
         inputDecls.push_back(initVals[initIndex++]);
         clockIndexes.push_back(curIndex);
-      } else {
-        inputDecls.push_back(rewriter.create<smt::DeclareFunOp>(loc, newTy));
+        continue;
       }
+      if (curIndex >= oldCircuitInputTy.size() - numRegs) {
+        auto initVal =
+            initialValues[curIndex - oldCircuitInputTy.size() + numRegs];
+        if (auto initIntAttr = dyn_cast<IntegerAttr>(initVal)) {
+          inputDecls.push_back(rewriter.create<smt::BVConstantOp>(
+              loc, initIntAttr.getValue().getSExtValue(),
+              dyn_cast<smt::BitVectorType>(newTy).getWidth()));
```

Comment:
If we are sure it can only be a BV we can to a `cast`, but don't we have to check if the type is a BV and error out otherwise? Are we doing this already somewhere?
It might also be possible to query the dialect's constant materializer.
```suggestion
              cast<smt::BitVectorType>(newTy).getWidth()));
```

Reply:
Good point! I'll try and swap to the constant materializer, that should be a lot nicer